### ISSUE TITLE
A way to programmatically set opacity and color of layers?

### ISSUE THREAD

**Issue by azatian** [scalableminds/webknossos-libs#827]
Hello!

Is there a way to programmatically set the opacity and the color of layers? Is there a value I can programmatically set within the meta data json file that would enable this automatically? Please see screenshot below. The default settings for these two layers is that there is 100 opacity and no color set.

Thank you for your help.

<img width="368" alt="Screen Shot 2022-11-28 at 2 14 41 PM" src="https://user-images.githubusercontent.com/9220290/204253066-80d23561-5aeb-4522-9e13-c6a42f6421ba.png">

**Comment by azatian** [comment]
I see! Is it possible to do this with wkcuber? I have been using wkcuber
CLI to convert our TIFF files into wkw then moving them into the
appropriate folder on our server where WK is hosted

**Comment by hotzenklotz** [comment]
> Is it possible to do this with wkcuber?

No, it is not possible with the wKCuber. You can, however, you use the wK Python Lib to convert a TIFF image stack with [`Dataset.from_images(...)`](https://docs.webknossos.org/api/webknossos/dataset/dataset.html#Dataset.from_images). This is used by the wkCuber under the hood anyway.

**Comment by azatian** [comment]
Thank you! I have been making some progress on this with the WK Python Lib
Each individual TIFF file has a shape of 160x160x16, but when I add the TIFFs as layers, the width comes out to be 160, the height comes out to be 16, and the depth comes out to be 1. Is there something that I am missing here?

`dataset = wk.Dataset("cutouts/dcvsyn1", voxel_size=(4,4,40))
   img_layer = dataset.add_layer_from_images(images="cutouts/dcvsyn1_source",
                                    layer_name="img", dtype='uint8', channel=0)

    pre_layer = dataset.add_layer_from_images(images="cutouts/dcvsyn1_presyn",
                                                layer_name="presyn", dtype='uint8', channel=0)

    post_layer = dataset.add_layer_from_images(images="cutouts/dcvsyn1_postsyn",
                                                layer_name="postsyn", dtype='uint8', channel=0)

    pre_layer.default_view_configuration = LayerViewConfiguration(
            color=(184, 66, 66), alpha=20)

    post_layer.default_view_configuration = LayerViewConfiguration(
            color=(57, 81, 137), alpha=20)`

**Comment by azatian** [comment]
I think I've somewhat figured out what might the issue be. Dataset.from_images reads the single but multi-page TIFF file differently from add_layer_from_images. Might have to do with Dataset.ConversionLayerMapping? See corresponding json file below.

```
dataset = wk.Dataset.from_images(input_path="cutouts/dcvsyn1_source",
                                output_path="cutouts/dcvsyn1",
                                voxel_size=(4,4,40),
                                name="dcvsyn1")
   
dataset.layers['vol.tiff'].name = "img"
    
pre_layer = dataset.add_layer_from_images(images="cutouts/dcvsyn1_presyn",
                                               layer_name="presyn", dtype='uint8', channel=1)
```




```
{
    "id": {
        "name": "dcvsyn1",
        "team": ""
    },
    "scale": [
        4.0,
        4.0,
        40.0
    ],
    "dataLayers": [
        {
            "name": "img",
            "category": "color",
            "boundingBox": {
                "topLeft": [
                    0,
                    0,
                    0
                ],
                "width": 160,
                "height": 160,
                "depth": 16
            },
            "elementClass": "uint8",
            "dataFormat": "wkw",
            "numChannels": 1,
            "wkwResolutions": [
                {
                    "cubeLength": 1024,
                    "resolution": [
                        1,
                        1,
                        1
                    ]
                }
            ]
        },
        {
            "name": "presyn",
            "category": "color",
            "boundingBox": {
                "topLeft": [
                    0,
                    0,
                    0
                ],
                "width": 160,
                "height": 16,
                "depth": 1
            },
            "elementClass": "uint8",
            "dataFormat": "wkw",
            "numChannels": 1,
            "wkwResolutions": [
                {
                    "cubeLength": 1024,
                    "resolution": [
                        1,
                        1,
                        1
                    ]
                }
            ]
        }
    ]
}
```

**Comment by jstriebel** [comment]
@azatian Thank you for reporting this bug! :lady_beetle:

I think I can reproduce the problem locally and will work on a fix. In the meantime it might work for you if you use the full path to the file in `add_layer_from_images`:
```python
dataset.add_layer_from_images(images="cutouts/dcvsyn1_presyn/name_of_your.tiff", …)
```

I'll write back when the fix is deployed, then it should work with the folder-name as well.

**Comment by azatian** [comment]
Hello! I see! I shall try that. I was able to get around it by doing making individual datasets per layer but tying them with the same output path: 

```
dataset = wk.Dataset.from_images(input_path="cutouts/dcvsyn1_source",
                                output_path="cutouts/dcvsyn1",
                                voxel_size=(4,4,40),
                                name="dcvsyn1")

dataset.layers['vol.tiff'].name = "img"
dataset_pre = wk.Dataset.from_images(input_path="cutouts/dcvsyn1_presyn",
                                output_path="cutouts/dcvsyn1",
                                voxel_size=(4,4,40),
                                name="dcvsyn1")
    

dataset_pre.layers['presyn.tiff'].default_view_configuration = LayerViewConfiguration(color=(184, 66, 66), alpha=20)
dataset_pre.layers['presyn.tiff'].name = "presyn"

dataset_post = wk.Dataset.from_images(input_path="cutouts/dcvsyn1_postsyn",
                                output_path="cutouts/dcvsyn1",
                                voxel_size=(4,4,40),
                                name="dcvsyn1")

 dataset_post.layers['postsyn.tiff'].default_view_configuration = LayerViewConfiguration(color=(57, 81, 137), alpha=20)
 dataset_post.layers['postsyn.tiff'].name = "postsyn"
```

**Comment by azatian** [comment]
Here is the associated json file, where the input shapes are now correct for all layers:

```
{
    "id": {
        "name": "dcvsyn1",
        "team": ""
    },
    "scale": [
        4.0,
        4.0,
        40.0
    ],
    "dataLayers": [
        {
            "name": "img",
            "category": "color",
            "boundingBox": {
                "topLeft": [
                    0,
                    0,
                    0
                ],
                "width": 160,
                "height": 160,
                "depth": 16
            },
            "elementClass": "uint8",
            "dataFormat": "wkw",
            "numChannels": 1,
            "wkwResolutions": [
                {
                    "cubeLength": 1024,
                    "resolution": [
                        1,
                        1,
                        1
                    ]
                }
            ]
        },
        {
            "name": "presyn",
            "category": "color",
            "boundingBox": {
                "topLeft": [
                    0,
                    0,
                    0
                ],
                "width": 160,
                "height": 160,
                "depth": 16
            },
            "elementClass": "uint8",
            "dataFormat": "wkw",
            "numChannels": 1,
            "defaultViewConfiguration": {
                "color": [
                    184,
                    66,
                    66
                ],
                "alpha": 20.0
            },
            "wkwResolutions": [
                {
                    "cubeLength": 1024,
                    "resolution": [
                        1,
                        1,
                        1
                    ]
                }
            ]
        },
        {
            "name": "postsyn",
            "category": "color",
            "boundingBox": {
                "topLeft": [
                    0,
                    0,
                    0
                ],
                "width": 160,
                "height": 160,
                "depth": 16
            },
            "elementClass": "uint8",
            "dataFormat": "wkw",
            "numChannels": 1,
            "defaultViewConfiguration": {
                "color": [
                    57,
                    81,
                    137
                ],
                "alpha": 20
            },
            "wkwResolutions": [
                {
                    "cubeLength": 1024,
                    "resolution": [
                        1,
                        1,
                        1
                    ]
                }
            ]
        }
    ]
}
```

**Comment by hotzenklotz** [comment]
@jstriebel Has this issue been resolved with PR #829? Perhaps, @azatian can re-test with your new release.

**Comment by jstriebel** [comment]
Closing this for now. @azatian please let us know if something is unresolved.
